src/Teams/Data/Repositories/ProgramCodeQuestionRepository.cs

```diff
@@ -18,5 +20,21 @@ public ProgramCodeQuestion PickById(Guid id)
         {
             return _db.ProgramCodeQuestions.Single(q => q.Id == id);
         }
+        public void SaveProgramText(ProgramCodeQuestion question, string text)
+        {
+            var model = new ProgramTextModel()
+            {
+                QuestionId = question.Id,
+                Program = text,
+                Status = 0
+            };
+            _db.ProgramTexts.Add(model);
+            _db.SaveChanges();
```

Comment:
Let's not save change in the repository. Otherwise, if we need to change multiple entities, we won't be able to store all changes in one transaction.

Reply:
In this case I should save changes in the controller, I suppose. Then I have to add IApplicationDbContext as a field of the controller, but there is no such method as "SaveChanges" in that interface. Moreover, there are no references to some DbSets, which are present in the ApplicationDbContext. So, should I extend this interface by adding all these things?

Reply:
Yeah, feel free to do that